src/main/java/com/seailz/discordjar/model/guild/Guild.java

```diff
@@ -1152,6 +1156,503 @@ public StringFormatter formatter() {
+                    RequestMethod.GET
+            ).invoke();
+            return req.body().getInt("pruned");
+        } catch (DiscordRequest.UnhandledDiscordAPIErrorException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Begins a prune operation on the guild. This requires the {@code KICK_MEMBERS} permission.
+     * @param days The amount of days to check users against, from 1-30.
+     */
+    public void prune(int days, List<Role> includedRoles) {
+        if (!Checker.inRange(1, 30, days, () -> Logger.getLogger("[DISCORD.JAR]").severe("Days cannot be outside 1-30!"))) return;
+
+        StringBuilder commaDelimitedSnowflakesString = new StringBuilder();
+        if (includedRoles.size() == 1) commaDelimitedSnowflakesString.append(includedRoles.get(0).id());
+        else {
+            for (Role r : includedRoles) {
+                commaDelimitedSnowflakesString.append(r.id());
+                commaDelimitedSnowflakesString.append(",");
+            }
+        }
+
+        try {
+            new DiscordRequest(
+                    new JSONObject()
+                            .put("days", days)
+                            .put("include_roles", commaDelimitedSnowflakesString),
+                    new HashMap<>(),
+                    URLS.POST.GUILDS.PRUNE.replace("{guild.id}", id),
+                    discordJar,
+                    URLS.POST.GUILDS.PRUNE,
+                    RequestMethod.POST
+            ).invoke();
+        } catch (DiscordRequest.UnhandledDiscordAPIErrorException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Returns a list of Guild {@link Member}s whose names (and optionally, nicknames) contain a filter string.
+     * @param filter The username filter string.
+     * @param includeNicknames Whether to include members whose nicknames contain the filter string.
+     * @return A list of Guild {@link Member}s
+     */
+    public List<Member> getMembersByName(String filter, boolean includeNicknames) {
+        try {
+            List<Member> namedMembers = new ArrayList<>();
+            for (Member member : getMembers()) {
+                if (includeNicknames) {
+                    if (member.user().username().contains(filter) || (member.nick() != null && member.nick().contains(filter))) namedMembers.add(member);
+                } else {
+                    if (member.user().username().contains(filter)) namedMembers.add(member);
+                }
+            }
+            return namedMembers;
+        } catch (DiscordRequest.UnhandledDiscordAPIErrorException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Kicks a {@link Member} from the Guild. This requires the {@code KICK_MEMBERS} permission.
+     * @param member The member to kick.
+     */
+    public void kickMember(Member member) {
+        try {
+            if (!discordJar.getIntents().contains(Intent.GUILD_MEMBERS)) {
```

Comment:
This does not require the intent. The REST API has no idea what intents you have enabled and therefore it can't act on them.

Reply:
resolved